src/components/EditorView.tsx

```diff
@@ -0,0 +1,329 @@
+        const localDecorations = decorations.find(
+          pos + offset + 1,
+          pos + offset + 1 + childNode.nodeSize
+        );
+        const inlineDecorations = localDecorations.filter(
+          (decoration) =>
+            (decoration as Decoration & { inline: boolean }).inline
+        );
+        const widgetDecorations = localDecorations.filter(
+          (decoration) =>
+            (decoration as Decoration & { type: ReactWidgetType })
+              .type instanceof ReactWidgetType
+        );
+        const textNodes: Node[] = makeCuts(
+          inlineDecorations
+            .flatMap((decoration) => [
+              decoration.from - (pos + offset + 1),
+              decoration.to - (pos + offset + 1),
+            ])
+            .concat(
+              widgetDecorations.map(
+                (decoration) => decoration.from - (pos + offset + 1)
+              )
+            ),
+          childNode
+        );
+
+        let subOffset = 0;
+        for (const textNode of textNodes) {
+          const textNodeStart = pos + offset + subOffset + 1;
+          const textNodeEnd = pos + offset + subOffset + 1 + textNode.nodeSize;
+          const marked = wrapInMarks(
+            <TextNodeWrapper pos={textNodeStart}>
+              {/* Text nodes always have text */}
+              {/* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */}
+              {textNode.text!}
+            </TextNodeWrapper>,
+            textNode.marks,
+            textNode.isInline
+          );
+          const decorated = wrapInDecorations(
+            marked,
+            inlineDecorations.filter(
+              (deco) => deco.from <= textNodeStart && deco.to >= textNodeEnd
+            ),
+            true
+          );
+
+          childElements.push(decorated);
+
+          widgetDecorations.forEach((decoration) => {
+            if (decoration.from !== textNodeEnd) return;
+
+            const decorationType = (
+              decoration as Decoration & { type: ReactWidgetType }
+            ).type;
+
+            childElements.push(<decorationType.Component />);
+          });
+
+          subOffset += textNode.nodeSize;
+        }
+
+        return false;
+      }
+
+      const outputSpec: DOMOutputSpec | undefined =
+        childNode.type.spec.toDOM?.(childNode);
+      if (!outputSpec)
+        throw new Error(
+          `Node spec for ${childNode.type.name} is missing toDOM`
+        );
+
+      const Component: ComponentType<NodeViewComponentProps> | undefined =
+        nodeViews[childNode.type.name];
+
+      let element: ReactNode = Component
+        ? createElement(Component, {
+            node: childNode,
+            decorations: [],
+            // TODO: This is how I'm reading the prosemirror-view code,
+            // but I would have expected this to be broader?
+            isSelected:
+              state.selection instanceof NodeSelection &&
+              state.selection.node === node,
```

Comment:
Yeah, I think that's right. We have used a broader definition at the Times in a custom way.